src/OrchardCore.Build/OrchardCore.Commons.props

```diff
@@ -13,7 +13,7 @@
     <GenerateAssemblyConfigurationAttribute>false</GenerateAssemblyConfigurationAttribute>
     <GenerateAssemblyCompanyAttribute>false</GenerateAssemblyCompanyAttribute>
     <GenerateAssemblyProductAttribute>false</GenerateAssemblyProductAttribute>
-
+    <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
```

Comment:
This will use the analyzer shipped with the .NET SDK, i.e. use the version that one has locally. This can cause inconsistent results, since analyzers change (a lot, you can see dozens of new violations in a previously pristine codebase with a new patch version even). A better way is to reference the analyzer package directly ([example](https://github.com/Lombiq/.NET-Analyzers/blob/5506480624315c9db6eccefe0a4097a81c357494/Lombiq.Analyzers/AnalyzerPackages.props#L10)).

However, `RunAnalyzersDuringBuild` ([see](https://learn.microsoft.com/en-us/visualstudio/code-quality/disable-code-analysis?view=vs-2022#net-framework-projects), and contrary the docs referring to .NET FW it also affects .NET) needs to be set to `false` by default, and overridden in the command in CI builds. This is so when you build locally, all the analyzers won't run for the whole solution (you'll still see the violations in the IDE for the currently open files, as well as during build for changed files).

Reply:
@Piedone I don't think I understand. I already removed the added `EnforceCodeStyleInBuild` then we build the project using this command `dotnet build -c Release -warnaserror /p:TreatWarningsAsErrors=true /p:RunAnalyzers=true /p:NuGetAudit=false`

so I would expect the the build to fail but instead it passes.